api/admin.go

```diff
@@ -374,14 +381,22 @@ func (a *API) adminUserCreate(w http.ResponseWriter, r *http.Request) error {
 	return sendJSON(w, http.StatusOK, user)
 }
 
-// adminUserDelete delete a user
+// adminUserDelete deletes a user
 func (a *API) adminUserDelete(w http.ResponseWriter, r *http.Request) error {
 	ctx := r.Context()
-	db := a.db.WithContext(ctx)
 	user := getUser(ctx)
 	adminUser := getAdminUser(ctx)
 
-	err := db.Transaction(func(tx *storage.Connection) error {
+	if user.IsSSOUser {
+		return badRequestError("user should be removed via identity provider instead")
+	}
+
+	var err error
+	params := &adminUserDeleteParams{}
+	q := r.URL.Query()
+	params.IsSoftDelete = strings.ToLower(q.Get("is_soft_delete"))
```

Comment:
`DELETE` requests can have bodies, so maybe a better idea is to move this to a JSON body?